string_manipulation.rb

```diff
@@ -1,30 +1,97 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  j = my_string.length - 1
+  while i < j do
+    temp = my_string[i]
+    my_string[i] = my_string[j]
+    my_string[j] = temp
+    i += 1
+    j -=1
+  end
+  return my_string
 end
 
 # A method to reverse each word in a sentence, in place.
 def reverse_words(my_words)
-  puts "NOT IMPLEMENTED"
+  index = 0
+  i = 0
+  while index < my_words.length
+    until my_words[index] == " " || (index == my_words.length)
+      index += 1
+    end
+    j = index - 1
+    while i < j do
+      temp = my_words[i]
+      my_words[i] = my_words[j]
+      my_words[j] = temp
+      i += 1
+      j -= 1
+    end
+    index = index + 1
+    i = index
+  end
+  return my_words
 end
 
 # A method to reverse the words in a sentence, in place.
 def reverse_sentence(my_sentence)
-  puts "NOT IMPLEMENTED"
+  string_reverse(my_sentence)
+  reverse_words(my_sentence)
+  return my_sentence
 end
 
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  j = my_phrase.length - 1
+  while i < j
+    if my_phrase[i] == " "
+      i += 1
+    elsif my_phrase[j] == " "

```

Comment:
This should be a separate if for checking if my_phrase[j] is a space. Even better to make both a while loop to account for consecutive white spaces.
`while my_phrase[i] == " "`
` i += 1`
`end`
Followed by a similar block for j.